gcs/CONFIGURATION.md

```diff
@@ -294,22 +298,39 @@ default service account impersonation.
         client cannot reliably write in the output stream from multiple threads
         without triggering *"Pipe broken"* exceptions;
 
-*   `fs.gs.outputstream.upload.chunk.size` (default: `67108864`)
+* `fs.gs.outputstream.pipe.buffer.size` (default: `1048576`)
+
+    Pipe buffer size used for uploading Cloud Storage objects. This pipe is an intermediate channel which is used to 
+    receive the data on one side and allow for reading of the data to be uploaded to GCS on the other side. 
+
+* `fs.gs.outputstream.upload.chunk.size` (default: `67108864`)
 
-    The number of bytes in one Google Cloud Storage upload request.
+    The number of bytes in one Google Cloud Storage upload request via the HTTP Media Uploader Service. 
+    This is used only for JSON API and for best performance should be a multiple of 8MB. 
+    
+    Having a large value example:- 64MB allows the upload performance must be much better due to fewer number of HTTP 
+    calls but on the other side if there are many partitions being written by the write call then each partition will hold 
+    64MB of data. Therefore for example:- if data for 250 partitions is written at once then the total memory requirement 
+    will be 250*64MB = 16 GB of memory resulting in OOM memory.
+ 
+    To arrive the optimal value this parameter needs to be tweaked based on the upload performance and number of concurrent files
+    being written.
+ 
+    More details can be found here 
+    https://cloud.google.com/java/docs/reference/google-api-client/latest/com.google.api.client.googleapis.media.MediaHttpUploader
 
-*   `fs.gs.outputstream.upload.cache.size` (default: `0`)
+* `fs.gs.outputstream.upload.cache.size` (default: `0`)
```

Comment:
Please, revert formatting, this doc is formatted with http://go/mdformat

Reply:
@medb - I have fixed the formatting. Please review